### PR TITLE
Add exit code flag

### LINKED ISSUE #58: Add --exit-code parameter
Add --exit-code parameter to exit with error status code on failure, to improve integration tests.
Inspired by [`git diff --exit-code`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code).

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR adds a new parameter to API to ensure plugin fails for any type of error.
This is a hidden convenience flag to test the plugin in a container.

Fixes #58 

## Proposed Changes

- Adds `--exit-code` flag
- Fixes duplicate flag error introduced by #68 changes
- Adds a new pipeline step to the drone flow, to dogfood our own pipeline.

### Description

- This PR adds a new parameter to API to ensure plugin fails for any type of error.
This is a hidden convenience flag to test the plugin in a container.

### Checklist

- [x] Read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] Read the [**CODE OF CONDUCT**](CODE_OF_CONDUCT.md) document.
- [x] Add tests to cover changes.
- [x] Ensure your code follows the code style of this project.
- [x] Ensure CI and all other PR checks are green OR
    - [x] Code compiles correctly.
    - [x] Created tests which fail without the change (if possible).
    - [x] All new and existing tests passed.
- [x] Add your changes to `Unreleased` section of [CHANGELOG](CHANGELOG.md).
- [x] Improve and update the [README](README.md) (if necessary).
- [x] Ensure [documentation](./DOCS.md) is up-to-date. The same file will be updated in [plugin index](https://github.com/drone/drone-plugin-index/blob/master/content/meltwater/drone-cache/index.md) when your PR is accepted, so it will be available for end-users at http://plugins.drone.io.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -319,7 +325,7 @@ func main() {
 			EnvVar: "SFTP_HOST",
 		},
 		cli.StringFlag{
-			Name:   "sftp-host",
+			Name:   "sftp-port",
```
Comment (kakkoyun) [inline main.go]: For the reviewer: This minor issue has introduced by the previous PR.

File: .drone.yml [hunk 2]
```diff
@@ -51,6 +51,37 @@ steps:
   environment:
     CGO_ENABLED: 0
 
+- name: release-snapshot-dev
```
Comment (kakkoyun) [inline .drone.yml]: For the reviewer: These newly added steps ensure `dev` tagged container up-to-date. That particular container is used to test the plugin on its on CI pipeline.

**Comment by kakkoyun** [comment]
@limayekaiwalya Have you had a chance to check this one?

**Comment by limayekaiwalya** [comment]
Not Yet @kakkoyun, I'll take a look tomorrow if that's okay.

**Comment by kakkoyun** [comment]
Hey @limayekaiwalya, I have a huge PR coming up, in order to prevent any merge conflicts, I'm going to go ahead and merge this one if you find any issues, please still comment and I'll address them in another PR.

**Pull request merged**